test/mapIndexed.js

```diff
@@ -0,0 +1,29 @@
+import * as R from 'ramda';
+
+import * as RA from '../src/index';
+import eq from './shared/eq';
+
+
+describe('mapIndexed', function () {
```

Comment:
Replicate all the tests that ramda uses for [map](https://github.com/ramda/ramda/blob/master/test/map.js).
Then add tests that on top of the example usage
```js
R.mapIndexed((val, idx, list) => idx + '-' + val, ['f', 'o', 'o', 'b', 'a', 'r']);
```
1.) Do the list equals the original list provided as an argument ?
2.) Cover the example above with unit test